File: revisiting_ctci/chapter2/KthToLast_2_2.go
/*
find the kth to last element of a singly linked list
*/

package main

import (
	"fmt"
	"./linkedlist"
)



func KthToLast(n linkedlist.Node, k int) linkedlist.Node {
	back := n.Back()

	for i := 1 ; i < k ; i ++ {
		back = *back.Prev
	}
	return back
}


func main(){
	data := linkedlist.Node{Data: 4}
	data.Add(6)
	data.Add(3)
	data.Add(2)
	data.Add(2)
	data.Add(2)
	data.Add(9)
	data.Add(1)
	data.Add(3)

	third_last := KthToLast(data,3)
	fmt.Println(data.Front())
	fmt.Println("The thrid last value is:")
	fmt.Println(third_last.Data)
}

File: revisiting_ctci/chapter2/KthToLast_2_2.py
from linkedlist import LinkedList, Node


def KthToLast(ll, n):
	ll.SetBack()

	i = 1
	while i < n:
		ll.MoveBackward()
		i += 1

	return ll





if __name__ == '__main__':

	n1 = 987654321

	ll1 = LinkedList()

	for n in str(n1):
		ll1.Add(int(n))

	print(ll1)
	x = KthToLast(ll1, 3)
	print(x.Data)

